### PR TITLE
Fix win32 breakage 18 dec

### PR DESCRIPTION
**Pull request by wrowe**
Description:
This corrects the newest breakage on the Windows platform introduced in December.

Risk Level: Low
Testing: Local on Windows msvc and Linux gcc
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: test/common/config/metadata_test.cc [hunk 1]
```diff
@@ -60,7 +60,7 @@ class TypedMetadataTest : public testing::Test {
 
   struct Bar : public TypedMetadata::Object {};
 
-  class FooFactory : public TypedMetadataFactory::TypedMetadataFactory {
+  class FooFactory : public Envoy::Config::TypedMetadataFactory {
```
Comment (jmarantz) [inline test/common/config/metadata_test.cc]: Why is this needed? It seems lexically obvious we are in a namespace where TypedMetadataFactory should be usable without qualification. If this cannot work I wonder what other things will fail?

File: test/common/common/mem_block_builder_test.cc [hunk 2]
```diff
@@ -71,25 +71,24 @@ TEST(MemBlockBuilderTest, AppendUint32) {
...
   MemBlockBuilder<uint8_t> mem_block(1);
   mem_block.appendOne(1);
-  EXPECT_DEATH({ mem_block.appendOne(2); },
 #ifdef ENVOY_CONFIG_COVERAGE
-               "" // For some reason, this test under coverage generates a list of testdata/*.
+  // For some reason, this test under coverage generates a list of testdata/*.
+  EXPECT_DEATH({ mem_block.appendOne(2); }, "");
 #else
-               ".*insufficient capacity.*"
+  EXPECT_DEATH({ mem_block.appendOne(2); }, ".*insufficient capacity.*");
```
Comment (jmarantz) [inline test/common/common/mem_block_builder_test.cc]: Thanks for doing this and sorry about this (this was mine). One minor suggestion:

```
#ifdef  ENVOY_CONFIG_COVERAGE
  static const char expected_death_regex[] = "";
#else 
  static const char expected_death_regex[] = ".*insufficient capacity.*";
#endif 
  EXPECT_DEATH({ mem_block.appendOne(2); }, expected_death_regex);
```
This seems better from a don't-repeat-yourself perspective. Here & below.

**Comment by jmarantz** [comment]
@zuercher for senior maintainer approval.

File: test/common/config/metadata_test.cc (hunk 1 above)
Comment (wrowe) [reply test/common/config/metadata_test.cc]: From this commit log;
> TypedMetadataFactory::TypedMetadataFactory remains too abstract, and didn't have base class methods name or parse.

We are quite confused as well, this is the error output of the original source;
```
ERROR: C:/workspace/envoy/test/common/config/BUILD:261:1: C++ compilation of rule '//test/common/config:metadata_test_lib_internal_only' failed (Exit 2)
test/common/config/metadata_test.cc(63): error C2516: 'Envoy::Config::TypedMetadataFactory::{ctor}': is not a legal base class
test/common/config/metadata_test.cc(65): error C3668: 'Envoy::Config::`anonymous-namespace'::TypedMetadataTest::FooFactory::name': method with override specifier 'override' did not override any base class methods
test/common/config/metadata_test.cc(68): error C3668: 'Envoy::Config::`anonymous-namespace'::TypedMetadataTest::FooFactory::parse': method with override specifier 'override' did not override any base class methods
test/common/config/metadata_test.cc(54): error C2664: 'Envoy::Registry::InjectFactory<Envoy::Config::TypedMetadataFactory>::InjectFactory(const Envoy::Registry::InjectFactory<Envoy::Config::TypedMetadataFactory> &)': cannot convert argument 1 from 'Envoy::Config::`anonymous-namespace'::TypedMetadataTest::FooFactory' to 'Base &'
        with
        [
            Base=Envoy::Config::TypedMetadataFactory
        ]
.\test/test_common/registry.h(17): note: see declaration of 'Envoy::Registry::InjectFactory<Envoy::Config::TypedMetadataFactory>::InjectFactory'
```

Things that are confusing, why promote TypedMetadataFactory:: to Envoy::Config:: in these contexts? Why the anon namespace? Suggestions and observations welcome.

File: test/common/common/mem_block_builder_test.cc (hunk 2 above)
Comment (wrowe) [reply test/common/common/mem_block_builder_test.cc]: Resolved as suggested, thanks!

File: test/common/config/metadata_test.cc (hunk 1 above)
Comment (zuercher) [reply test/common/config/metadata_test.cc]: Does it work if you just extend `TypedMetadataFactory` instead of `TypedMetadataFactory::TypedMetadataFactory`? The latter seems to be nonsense, since there's no TypedMetadataFactory namespace in the code.

File: tools/clang_tools/api_booster/main.cc [hunk 3]
```diff
@@ -112,7 +112,7 @@ class ApiBooster : public clang::ast_matchers::MatchFinder::MatchCallback,
     return true;
   }
 
-  // Visitor callback for end of a compiation unit.
+  // Visitor callback for end of a compilation unit.
```
Comment (zuercher) [inline tools/clang_tools/api_booster/main.cc]: Can you remove "compiation" from the dictionary.

File: tools/clang_tools/api_booster/main.cc [hunk 4]
```diff
@@ -161,7 +161,7 @@ class ApiBooster : public clang::ast_matchers::MatchFinder::MatchCallback,
       // Die hard if we don't have a useful proto type for something that looks
       // like an API type(modulo a short whitelist).
       std::cerr << "Unknown API type: " << proto_type_name << std::endl;
-      // TODO(htuch): mabye there is a nicer way to terminate AST traversal?
+      // TODO(htuch): maybe there is a nicer way to terminate AST traversal?
```
Comment (zuercher) [inline tools/clang_tools/api_booster/main.cc]: Likewise.

File: test/common/config/metadata_test.cc (hunk 1 above)
Comment (wrowe) [reply test/common/config/metadata_test.cc]: @zuercher agreed, that was the issue. Will push the fix first thing Friday.

**Comment by wrowe** [comment]
The OS/X failure to grab bazelisk-darwin-amd64 is unrelated to this commit, can you re-review this PR please, @zuercher?
